tools/alara_output_processing/alara_output_plotting.py

```diff
@@ -0,0 +1,285 @@
+
+def preprocess_data(df_dicts, data_key='Data', seconds=True):
+    '''
+    Extract all of the ALARADFrames from df_dicts, their (shared) time
+        columns and store them in a list of tuples for each plot to be
+        produced. Additionally, for each plot, establish a corresponding
+        colormap.
+
+    Arguments:
+        df_dicts (dict or list): Single dictionary containing an ALARA output 
+            ALARADFrame and its metadata, of the form:
+            df_dict = {
+                'Run Label'   : (Distinguisher between runs),
+                'Variable'    : (Any ALARA output variable, dependent on ALARA
+                                 run parameters),
+                'Unit'        : (Respective unit matching the above variable),
+                'Data'        : (ALARADFrame containing ALARA output data for
+                                 the given run parameter, variable, and unit)
+            }
+        data_key (str, optional): Keyword for df_dicts to direct to the
+            desired ALARADFrame to source the plot's data.
+            (Defaults to 'Data')
+        seconds (bool, optional): Option to convert cooling times from
+            years to seconds.
+            (Defaults to True)
+
+    Returns:
+        all_data (list of tuples): A list of all relevant data for each table
+            to be plotted including:
+                df_dict (dict): Single dictionary from list of dictionaries in
+                    input df_dicts.
+                adf (alara_output_processing.ALARADFrame): Specialized ALARA
+                    output DataFrame containing the extracted tabular data for
+                    a single variable and interval/zone of an ALARA run.
+                times (list of floats): Cooling times (columns of adf).
+    '''
+
+    all_labels = set()
+    all_data = []
+    for df_dict in df_dicts:
+        adf = df_dict[data_key]
+        times = adf.process_time_vals(seconds=seconds)
+        adf = adf.T
+        for col in adf.columns:
+            label_text = f"{adf[col].iloc[0]}"
+            all_labels.add(label_text)
+
+        all_data.append((df_dict, adf, times))
+
+    labels_sorted = sorted(all_labels)
+
+    cmap = plt.cm.get_cmap('Dark2')
+    color_map = {lbl: cmap(i % cmap.N) for i, lbl in enumerate(labels_sorted)}
+
+    return all_data, color_map
+
+def split_label(label):
+    '''
+    Split the string of a series' label to extract the isotope being plotted
+        and (conditionally) the run label attached to it. The run parameter
+        will be present parenthetically if two runs are being plotted
+        comparitavely.
+    Arguments:
+        label (str): Series label generated from 
+            ax.get_legend_handles_labels().
+    
+    Returns:
+        isotope (str): Isotope being plotted.
+        run_lbl (str): Run label of plotted isotope. Will be empty string
+            for single run plotting.
+    '''
+
+    if '(' in label:
```

Comment:
Can you add a comment about the assumed format of the label if this condition is true?